das/state.go

```diff
@@ -77,17 +81,18 @@ func (s *coordinatorState) updateHead(last uint64) bool {
 		return false
 	}
 
-	if s.networkHead == genesisHeight {
+	// TODO(@derrandz): question to @walldiss, why is networkHead uint64 to begin with?
```

Comment:
TODO to be resolved

Reply:
It is `uint64` so it is possible to compare it with other variables of same type without conversion.

Reply:
@walldiss I am trying to understand which variable is the one that has to be `unit64` and that caused the rest to become `uint64` to reduce conversion.

Previously in this PR, you mentioned https://github.com/celestiaorg/celestia-node/pull/1225#discussion_r998376041 and https://github.com/celestiaorg/celestia-node/pull/1225#discussion_r998406090 which have lead me to believe that `networkHead` was the original variable that had to be `uint64` (_for some reason_) and thus caused the rest to become so to avoid conversion.

If there is no particular reason to using `uint64` specifically, let's revert to `uint`.

Reply:
Thare are many of them, `uint64` is used for any block heights variables inside DASer.  `uint64` is used over `uint` to avoid usage of uint32 on 32bit platforms.

Reply:
As per our conversation, I will revert everything in daser to uint with the argument that:

No concrete argument to raise precision to uint64 or other.
We only know so far that we need positive values for the concerned variables.
:+1: